demo/contracts/contracts/DataStreamsConsumer.sol

```diff
@@ -61,302 +28,69 @@ contract DataStreamsConsumer is
         string feedId;
     }
 
-    struct Quote {
-        address quoteAddress;
-    }
-
-    // ================================================================
-    // |                          Events                              |
-    // ================================================================
-
-    event InitiateTrade(
-        address msgSender,
-        address tokenIn,
-        address tokenOut,
-        uint256 amountIn,
-        string feedId
-    );
-
     event TradeExecuted(uint256 tokensAmount);
 
-    // ================================================================
-    // |                          Errors                              |
-    // ================================================================
-
-    error InvalidFeedId(string feedId);
-
-    /**
-     * @dev Initializes the contract with necessary parameters.
-     * @param router The address of the swap router contract.
-     * @param verifier The address of the verifier contract.
-     * @param linkToken The address of the LINK token contract.
-     * @param feedsHex An array of hexadecimal feed IDs.
-     */
-
     function initializer(
         address router,
-        address payable verifier,
-        address linkToken,
-        string[] memory feedsHex
+        address oracleEmitter
     ) public {
         i_router = ISwapRouter(router);
-        i_verifier = IVerifierProxy(verifier);
-        i_linkToken = linkToken;
-        s_feedsHex = feedsHex;
+        i_oracleEmitter = IOracle(oracleEmitter);
     }
 
-    // ================================================================
-    // |                        Chainlink DON                          |
-    // ================================================================
+    function randomUINT() internal view returns (uint256) {
```

Comment:
it will return the same value during block, block might have more than one trade transaction